### ISSUE TITLE
URI set to localhost:0 when a random port is used.

### ISSUE THREAD

**Issue by grunenwflorian** [hanleyt/jersey-junit#8]
Hey Tomas,

Whenever we use the Jersey Extension to create a server using a random port (by setting "jersey.config.test.container.port" to "0" by example) the URI injected is set to "http://localhost:0".

The reason is that the URI object is added to the store before starting the server.
```java 
            @Override
            protected Application configure() {
                getStore(context).put(URI.class, getBaseUri());
                return applicationProvider.apply(context);
            }
```

I think moving that here : 
```java 
    @Override
    public void beforeEach(ExtensionContext context) throws Exception {
        JerseyTest jerseyTest = initJerseyTest(context);
       // added 
       getStore(context).put(URI.class, jerseyTest.target().getUri());
      //
        getStore(context).put(Client.class, jerseyTest.client());
        getStore(context).put(WebTarget.class, jerseyTest.target());
    }
```

will fix it.

On an unrelated note : 
Would you consider removing the jersey-test-framework-provider-grizzly2 dependency ?
That would allow us to pick the type of provider more easily. Currently we are considering the jetty one.

If you are okay with one of those changes I'd gladly create a PR, just le me know.

Thanks.

**Comment by hanleyt** [comment]
Hey Florian,
Yeah that all sounds good to me, a PR would be more than welcome!
Thanks
Tomas

**Comment by hanleyt** [comment]
Thanks for the PR @grunenwflorian!
Merged and release 2.0.0 created: https://github.com/hanleyt/jersey-junit/releases/tag/2.0.0
